src/plugins/intel_cpu/tests/functional/subgraph_tests/src/stateful_model.cpp

```diff
@@ -0,0 +1,547 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <common_test_utils/ov_tensor_utils.hpp>
+#include "shared_test_classes/base/ov_subgraph.hpp"
+#include "ov_models/utils/ov_helpers.hpp"
+#include "ov_models/builders.hpp"
+
+using namespace InferenceEngine;
+using namespace ov::test;
+
+namespace SubgraphTestsDefinitions {
+
+class StatefulModelTest : public SubgraphBaseTest {
+public:
+    static constexpr ov::element::Type_t testPrc = ov::element::Type_t::f32;
+
+public:
+    void prepare() {
+        compile_model();
+        inferRequest = compiledModel.create_infer_request();
+        ASSERT_TRUE(inferRequest);
+    }
+
+    void reset_state() {
+        for (auto&& state : inferRequest.query_state()) {
+            state.reset();
+        }
+    }
+
+    static void float_compare(const float* expected_res, const float* actual_res, size_t size) {
+        constexpr float rel_diff_threshold = 1e-4f;
+        for (size_t i = 0; i < size; ++i) {
+            const float expected_val = expected_res[i];
+            const float actual_val = actual_res[i];
+            if (0.f == expected_val) {
+                ASSERT_TRUE(abs(actual_val) < rel_diff_threshold);
+            } else {
+                ASSERT_TRUE(abs(actual_val / expected_val - 1.f) < rel_diff_threshold);
+            }
+        }
+    }
+};
+
+//     ┌────────┐     ┌───────┐
+//     │ Param1 │     │ Const │
+//     └───┬────┘     └───┬───┘
+//         │              │
+//         │         ┌────┴──────┐
+//  .......│.........│ ReadValue │
+//  .      │         └────┬─────┬┘
+//  .      │              │     │
+//  .      │   ┌─────┐    │     │
+//  .      └───┤ Add ├────┘     │
+//  .          └──┬──┘          │
+//  .             │             │
+//  .             │             │
+//  . ┌────────┐  │    ┌─────┐  │
+//  ..│ Assign ├──┴────┤ Add ├──┘
+//    └────────┘       └──┬──┘
+//                        │
+//                        │
+//                   ┌────┴──────┐
+//                   │  Result   │
+//                   └───────────┘
+
+class StaticShapeStatefulModel : public StatefulModelTest {
+public:
+    void SetUp() override {
+        targetDevice = ov::test::utils::DEVICE_CPU;
+        ov::element::Type netPrc = testPrc;
+
+        const ov::Shape inpShape = {1, 1};
+        targetStaticShapes = {{inpShape}};
+
+        auto arg = std::make_shared<ov::op::v0::Parameter>(netPrc, ov::Shape{1, 1});
+        auto init_const = ov::op::v0::Constant::create(netPrc, ov::Shape{1, 1}, {0});
+
+        // The ReadValue/Assign operations must be used in pairs in the model.
+        // For each such a pair, its own variable object must be created.
+        const std::string variable_name("variable0");
+        auto variable = std::make_shared<ov::op::util::Variable>(
+            ov::op::util::VariableInfo{ov::PartialShape::dynamic(), ov::element::dynamic, variable_name});
+
```

Comment:
According to PR #20404 , the output shape will be [got](https://github.com/openvinotoolkit/openvino/blob/6adc6064b0c0340dade3844be89710d9a92c9f0c/src/core/src/op/read_value.cpp#L62) from the state instead of `init_cost`, then a dynamic rank node will be created and the cpu plugin may raise error.

Reply:
Yep. With no doubts it will do that. Thus, depending on which PR is merged earlier, some changes need to be introduced.